cdap-common/src/main/java/co/cask/cdap/common/namespace/AbstractNamespaceClient.java

```diff
@@ -100,13 +104,46 @@ public void create(NamespaceMeta namespaceMeta)
     if (response.getResponseCode() == HttpURLConnection.HTTP_BAD_REQUEST) {
       throw new BadRequestException("Bad request: " + responseBody);
     }
-    throw new IOException(String.format("Cannot create namespace %s. Reason: %s", namespaceMeta.getName(), response));
+    throw new IOException(String.format("Cannot create namespace %s. Reason: %s", metadata.getName(), response));
+  }
+
+  @Override
+  public boolean exists(Id.Namespace namespaceId) throws Exception {
+    return get(namespaceId) != null;
+  }
+
+  @Override
+  public void updateProperties(Id.Namespace namespaceId, NamespaceMeta metadata) throws Exception {
+    URL url = resolve(String.format("namespaces/%s/properties", namespaceId.getId()));
+    HttpResponse response = execute(HttpRequest.put(url).withBody(GSON.toJson(metadata)).build());
+    String responseBody = response.getResponseBodyAsString();
+    if (response.getResponseCode() == HttpURLConnection.HTTP_OK) {
+      return;
+    }
+    if (response.getResponseCode() == HttpURLConnection.HTTP_BAD_REQUEST) {
+      throw new BadRequestException("Bad request: " + responseBody);
+    }
+    throw new IOException("Cannot get create namespace. Reason: " + "getDetails(response)");
+  }
+
+  @Override
+  public void deleteDatasets(Id.Namespace namespaceId) throws Exception {
+    URL url = resolve(String.format("unrecoverable/namespaces/%s/datasets", namespaceId.getId()));
+    HttpResponse response = execute(HttpRequest.delete(url).build());
+
+    if (response.getResponseCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+      throw new NamespaceNotFoundException(namespaceId);
+    } else if (HttpURLConnection.HTTP_FORBIDDEN == response.getResponseCode()) {
+      throw new NamespaceCannotBeDeletedException(namespaceId, response.getResponseBodyAsString());
+    } else if (response.getResponseCode() == HttpURLConnection.HTTP_OK) {
+      return;
+    }
+    throw new IOException("Cannot delete namespace. Reason: " + response.getResponseBodyAsString(Charsets.UTF_8));
```

Comment:
It's fine to just omit the parameter, if its the default value? (for consistency)